### PR TITLE
layers: Improve C/C++ std error checking

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5213
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5212

### CODE REVIEW

**Review by artem-lunarg — APPROVED** [review]

**Review by juan-lunarg — APPROVED** [review]

File: tests/vklayertests_instanceless.cpp [hunk 1]
```diff
@@ -189,6 +189,10 @@ TEST_F(VkLayerTest, InstanceBadValidationFlags) {
 void* VKAPI_PTR DummyAlloc(void*, size_t size, size_t alignment, VkSystemAllocationScope) {
     size_t space = size + alignment - 1;
     void* mem_ptr = std::malloc(space);
+    if (!mem_ptr) {
+        printf("Dummy Alloc failed\n");
```
Comment (elfring) [inline tests/vklayertests_instanceless.cpp:193]: :thought_balloon: It can be nicer to use the function “perror”, can't it?

Comment (jeremyg-lunarg) [reply tests/vklayertests_instanceless.cpp:193]: We do accept pull requests.

File: tests/vktestframework.cpp [hunk 2]
```diff
@@ -690,7 +690,10 @@ char **VkTestFramework::ReadFileData(const char *fileName) {
 
     while (fgetc(in) != EOF) count++;
```
Comment (elfring) [inline tests/vktestframework.cpp:691]: :thought_balloon: Is the error detection and corresponding still improvable also according to this source code?

**Review by elfring — COMMENTED** [review]
Why were changes for different issues mixed in such a commit?

**Pull request merged**